### ISSUE TITLE
actix-web does not parse the repetitions in regex correctly

### ISSUE THREAD

**Issue by uzytkownik** [actix/actix-web#500]
```rust
.resource("/pull/{id:[[:xdigit:]]{2}}", |r| r.with_async(pull))
```

Results in errors:

```
thread 'arbiter:6064e037-e699-4972-b47a-4344a7069b05:actor' panicked at 'Wrong path pattern: "/pull/{id:[[:xdigit:]]{2}}" regex parse error:
    ^/pull/(?P<id>]]{2)\}$
                    ^^
error: unclosed counted repetition', /home/mpiechotka/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-web-0.7.5/src/router.rs:578:29
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

**Comment by fafhrd91** [comment]
Would you provide pr?

**Comment by fafhrd91** [comment]
fixed by #503
